### ISSUE TITLE
Add Entware-ng

### ISSUE THREAD

**Issue by ryzhovau** [repology/repology#435]
Please, add [Entware-ng](https://github.com/Entware-ng/Entware-ng) repo. There is OpenWrt-compatible indexes at:

* http://pkg.entware.net/binaries/mipsel/Packages.gz
* http://pkg.entware.net/binaries/armv5/Packages.gz
* http://pkg.entware.net/binaries/armv7/Packages.gz
* http://pkg.entware.net/binaries/x86-32/Packages.gz
* http://pkg.entware.net/binaries/x86-64/Packages.gz

Can't say our own [HTML indexer](http://pkg.entware.net/binaries/mipsel/Packages.html) looks pretty enough:)

**Comment by AMDmi3** [comment]
This is doable the same way as OpenWRT/LEDE are supported, but be aware that this support is incomplete. Embedded specifics result in a lot of package splits which in turn require a lot of merging rules to be properly compared to other repos. For now there are no such rules and packages which don't match with other repos are hidden, so you'll only see a fraction of packages in repology.

Alternative solution would be to use source packages equivalent for these repos (I think I've seen something for LEDE, but it was hard to parse). It will also fix the unwanted necessity to parse a lot of arch-specific repositories.

Is there something like that for this repo? Also, is it related to OpenWRT/LEDE in any way?

**Comment by ryzhovau** [comment]
> Is there something like that for this repo? 

Our sources are compatible with OpenWrt/LEDE Buildroot. 
> Also, is it related to OpenWRT/LEDE in any way?

While OpenWrt/LEDE is a separate firmware with their own repo, Entware-ng packages are firmware independent environment for routers/NAS/media players. 3/4 of packages are taken form OpenWrt, the rest 1/4 is unique.

**Comment by AMDmi3** [comment]
So is there a way to get source packages?

**Comment by ryzhovau** [comment]
There's no any source links in `Packages.gz` index, so, you need to parse every Makefile form [here](https://github.com/openwrt/packages) to extract `PKG_NAME` and `PKG_VERSION` from there:(

**Comment by AMDmi3** [comment]
Thought so... Ok, I'll add it the same way openwrt/lede are supported for now this week. Thanks for submission!

**Comment by ryzhovau** [comment]
Also, keep in mind of [basic OpenWrt packages](https://github.com/openwrt/openwrt/tree/master/package) outside of feeds.

**Comment by AMDmi3** [comment]
Sorry for delay, deploying tomorrow.

**Comment by AMDmi3** [comment]
@ryzhovau btw, I'm working on bringing back OpenWRT support to Repology right now, and I've discovered that there's (now) a `Packages.manifest` file (https://downloads.openwrt.org/releases/18.06.1/packages/x86_64/base/Packages.manifest), which is almost the same as `Packages`, but it also contains `Source` and `Maintainer` fields. While the latter is just useful, the former is crucial to proper repository support, as it allows to automatically gather numerous subpackages into project entities which Repology can operate on.

I wonder if something similar could be made available for Entware repositories.

**Comment by ryzhovau** [comment]
Done via https://github.com/Entware/ipk-html-indexer/commit/aa15d2d0d9e5305a2d3f829f8f141897dff4dfb8
